packages/tracing/src/transaction.ts

```diff
@@ -227,69 +229,43 @@ export class Transaction extends SpanClass implements TransactionInterface {
    *
    * @experimental
    */
-  public getBaggage(): Baggage {
-    const existingBaggage = this.metadata.baggage;
-
-    // Only add Sentry baggage items to baggage, if baggage does not exist yet or it is still
-    // empty and mutable
-    const finalBaggage =
-      !existingBaggage || isBaggageMutable(existingBaggage)
-        ? this._populateBaggageWithSentryValues(existingBaggage)
-        : existingBaggage;
-
-    // Update the baggage stored on the transaction.
-    this.metadata.baggage = finalBaggage;
-
-    return finalBaggage;
-  }
+  public getDynamicSamplingContext(): Readonly<Partial<DynamicSamplingContext>> | undefined {
+    if (this._frozenDynamicSamplingContext) {
+      return this._frozenDynamicSamplingContext;
+    }
 
-  /**
-   * Collects and adds data to the passed baggage object.
-   *
-   * Note: This function does not explicitly check if the passed baggage object is allowed
-   * to be modified. Implicitly, `setBaggageValue` will not make modification to the object
-   * if it was already set immutable.
-   *
-   * After adding the data, the baggage object is set immutable to prevent further modifications.
-   *
-   * @param baggage
-   *
-   * @returns modified and immutable baggage object
-   */
-  private _populateBaggageWithSentryValues(baggage: Baggage = createBaggage({})): Baggage {
     const hub: Hub = this._hub || getCurrentHub();
     const client = hub && hub.getClient();
 
-    if (!client) return baggage;
+    if (!client) return undefined;
```

Comment:
Since we have the `if (Object.keys(object).length === 0) {` check in `objectToBaggageHeader`, I think we can get away with returning an empty object here - but I'll leave it up to you to make the call.

Reply:
You're right. I like it because it allows us to simplify things a bit! 55aa327a905abbb4e4d8f104de80c337f866d506